File: code/get_TreeHeight.R
# This function calculates heights of trees given distance of each tree 
# from its base and angle to its top, using  the trigonometric formula 
#
# height = distance * tan(radians)
#
# ARGUMENTS
# degrees:   The angle of elevation of tree
# distance:  The distance from base of tree (e.g., meters)
#
# OUTPUT
# The heights of the tree, same units as "distance"

# Load in the dataset, given as an commad line argument 
args<-commandArgs(TRUE)

Data<- read.csv(paste0("../data/", args, ".csv"), stringsAsFactors = F)


# Make a new column and fill with NA values
Data$Rad <- NA

# define radians as degrees*pi/180
###################################
# for each row (1:nrow(TreeData)) in the dataset:
# For each observation in the third column [i, 3], multiply by pi/180 and  then plug these values into "Rad" column [i,4]
for (i in 1:nrow(Data)){
  
  #  browser() ## use this breakpoint to see what each loop does
  Data[i,4] <- Data[i,3] * pi/180
} 

# define height as distance*ran(radians)

# For every 1 to x many number of rows of the "TreeData" dataset
# make new column called "Height.n", and for each iteration, take values from the second column of the the dataset and multiply by tan(radians) and put into new column (Tree.Height.m)
for (i in 1:nrow(Data)){ 
  Data$Tree.Height.n[i] <- Data[i,2] * tan(Data$Rad[i])
}
#note: should make column first, but can also just use $ to add new column 

############
# longer way
############
#for (i in (TreeData$Distance.m)){ 
#  height <- i * tan(radians)
#  TreeData$Tree.Height.n[i] <- height
#} 

# Export TreeDta as a .csv and save in results directory 
write.csv(Data, paste0("../results/", args, "_treeheight.csv"))

# Script complete
print("Script complete! :)")



# Note- this works if you strip extension (.csv) and file path (../results) from  command line argument,  but check if this is okay or do we need to strip extension and filepath from inside script. 

File: code/get_TreeHeight.py
"""get_TreeHeight is a python script that calculates tree heights"""

<<<<<<< HEAD
__appname__ = 'get_TreeHeight.py'
__author__ = 'Peter Zeng (Email: pz221@ic.ac.uk)'
=======
__appname__ = 'get_TreeHeight.py' 
__author__ = 'Peter (Guancheng) Zeng (pz221@ic.ac.uk), Kate Griffin (kate.griffin21@imperial.ac.uk), Junyue Zhang (jz1621@ic.ac.uk), Kayleigh Greenwood (Kayleigh.Greenwood21@ic.ac.uk)'
>>>>>>> 018a50b17243a8b2c0adeea902645bafee4e208c
__version__ = '0.0.1'


### import ###
import sys
import pandas
import numpy

treeData = pandas.read_csv("../data/" + sys.argv[1] + ".csv")

### to calculate the height for every rows ###
for row in treeData:
    treeData["Tree.Height.m"] = treeData["Distance.m"] * numpy.tan(treeData["Angle.degrees"] * numpy.pi/180)

### output the data to the tree data csv
<<<<<<< HEAD
treeData.to_csv("../results/" + sys.argv[1] + "_TreeHts_py.csv", sep= ',',index=False)
=======
treeData.to_csv("../results/" + sys.argv[1] + "_TreeHts_py.csv", sep= ',',index=False)
>>>>>>> 018a50b17243a8b2c0adeea902645bafee4e208c
